src/pyhf/infer/test_statistics.py

```diff
@@ -47,16 +90,135 @@ def qmu(mu, data, pdf, init_pars, par_bounds):
-    )
-    qmu = fixed_poi_fit_lhood_val - unconstrained_fit_lhood_val
-    qmu = tensorlib.where(
-        muhatbhat[pdf.config.poi_index] > mu, tensorlib.astensor(0.0), qmu
-    )
-    return tensorlib.clip(qmu, 0, max_value=None)
+
+def qmu_tilde(mu, data, pdf, init_pars, par_bounds):
+    r"""
+    The test statistic, :math:`\tilde{q}_{\mu}`, for establishing an upper
+    limit on the strength parameter, :math:`\mu` for models with
+    bounded POI.
+
+    Example:
+        >>> import pyhf
+        >>> pyhf.set_backend("numpy")
+        >>> model = pyhf.simplemodels.hepdata_like(
+        ...     signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
+        ... )
+        >>> observations = [51, 48]
+        >>> data = pyhf.tensorlib.astensor(observations + model.config.auxdata)
+        >>> test_mu = 1.0
+        >>> init_pars = model.config.suggested_init()
+        >>> par_bounds = model.config.suggested_bounds()
+        >>> pyhf.infer.test_statistics.qmu_tilde(test_mu, data, model, init_pars, par_bounds)
+        array(3.93824492)
+
+    Args:
+        mu (Number or Tensor): The signal strength parameter
+        data (Tensor): The data to be considered
+        pdf (~pyhf.pdf.Model): The statistical model adhering to the schema model.json
+        init_pars (`list`): Values to initialize the model parameters at for the fit
+        par_bounds (`list` of `list`\s or `tuple`\s): The extrema of values the model parameters are allowed to reach in the fit
+
+    Returns:
+        Float: The calculated test statistic, :math:`\tilde{q}_{\mu}`
+    """
+    if pdf.config.poi_index is None:
+        raise UnspecifiedPOI(
+            'No POI is defined. A POI is required for profile likelihood based test statistics.'
+        )
+    if par_bounds[pdf.config.poi_index][0] != 0:
+        log.warning(
+            'qmu_tilde test statistic used for fit configuration with POI not bounded at zero.\n'
+            + 'Use the qmu test statistic (pyhf.infer.test_statistics.qmu) instead.'
+        )
+    return _qmu_like(mu, data, pdf, init_pars, par_bounds)
+
+
+def tmu(mu, data, pdf, init_pars, par_bounds):
+    r"""
+    The test statistic, :math:`t_{\mu}`, for establishing a two-sided
+    interval on the strength parameter, :math:`\mu`.
+
+    Example:
+        >>> import pyhf
+        >>> pyhf.set_backend("numpy")
+        >>> model = pyhf.simplemodels.hepdata_like(
+        ...     signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
+        ... )
+        >>> observations = [51, 48]
+        >>> data = pyhf.tensorlib.astensor(observations + model.config.auxdata)
+        >>> test_mu = 1.0
+        >>> init_pars = model.config.suggested_init()
+        >>> par_bounds = model.config.suggested_bounds()
+        >>> par_bounds[model.config.poi_index] = [-10.0, 10.0]
+        >>> pyhf.infer.test_statistics.tmu(test_mu, data, model, init_pars, par_bounds)
+        3.954989104752883
```

Comment:
At the moment the `tmu` and `tmu_tilde` test stats are returning pure floats instead of 0-dimensional tensors like the `qmu` and `qmu_tilde` test stats. There should probably be harmonized (to 0-dim tensors?).

Reply:
hm i'm curious how that happens.. any idea? qmu in principle is just a wrapper over tmu

Reply:
My guess was the `tensorlib.where` call but can't figure it out

```python
>>> import numpy as np
>>> np.where(True, np.asarray(1.0), np.asarray(-1.0))
array(1.)
>>> np.where(np.asarray(True), np.asarray(1.0), np.asarray(-1.0))
array(1.)
>>> np.where(np.asarray(True), 1.0, np.asarray(-1.0))
array(1.)
>>> np.where(True, 1.0, np.asarray(-1.0))
array(1.)
>>> np.where(False, 1.0, np.asarray(-1.0))
array(-1.)
```

Reply:
clip will also operate on floats, and so was returning a float instead of a 0-d tensor.